src/libsyntax/parse/lexer/mod.rs

```diff
@@ -1445,7 +1445,7 @@ impl<'a> StringReader<'a> {
                                     format!("\"{}\"", &self.src[start..end]),
                                     Applicability::MachineApplicable
                                 ).emit();
-                            return Ok(token::Literal(token::Str_(Symbol::intern("??")), None))
+                            FatalError.raise();
```

Comment:
I believe this will cause the compiler to stop working at the parser stage, we would prefer to continue checking types and lifetimes in the face of recoverable errors, like this would be.